### PR TITLE
Show correct TVL for Beanstalk (BEAN)

### PR DESCRIPTION
**Pull request by Taimonania**
##### Current TVL:
152.78 M as of block 14590383.

##### Chain:
ethereum

##### methodology (what is being counted as tvl, how is tvl being calculated):
Short: Counts all beans and current LPs in the silo.

Long: Calculates currently locked Beans, locked BEAN:ETH Uniswap LP and locked BEAN:3CRV LP and BEAN:LUSD LP on Curve and unwraps underlying tokens to get current TVL.

### CODE REVIEW

**Comment by llamatester** [comment]
The adapter at projects/bean/index.js exports TVL: 
        
 
 
    ethereum                  152.75 M
    
    total                    152.75 M

File: projects/bean/index.js [hunk 1]
```diff
@@ -1,12 +1,74 @@
...
+]
 
 async function tvl(time, block){
-    const balances = {}
-    await sumTokens(balances, [["0xDC59ac4FeFa32293A95889Dc396682858d52e5Db", "0xC1E088fC1323b20BCBee9bd1B9fC9546db5624C5"]], block)
-    return balances
+    const balances = {};
+
+    const beanBalance = (await sdk.api.abi.call({
+        abi: bean_abi["totalDepositedBeans"],
```
Comment (g1nt0ki) [inline projects/bean/index.js:22]: can this be replaced with


      const beanBalance = (await sdk.api.erc20.balanceOf({
        chain: 'ethereum',
        target: BEAN_TOKEN_ADDR,
        owner: BEAN_DIA_ADDR,
        block,
    })).output;
    
  Also, can this be moved to staking?

File: projects/bean/index.js [hunk 2]
```diff
@@ -1,12 +1,74 @@
...
+        chain: 'ethereum',
+        target: BEAN_DIA_ADDR,
+        block: block,
+    })).output;
+
+    const beanEthLpBalance = (await sdk.api.abi.call({
```
Comment (g1nt0ki) [inline projects/bean/index.js:28]: same, can you count here using `erc20.balanceOf` or you could use `sumTokensAndLPsSharedOwners` from `unwrapLPs` helper

Can you move this value to pool2

File: projects/bean/index.js [hunk 3]
```diff
@@ -1,12 +1,74 @@
...
+    await sdk.util.sumSingleBalance(balances, BEAN_TOKEN_ADDR, beanBalance);
+    // add balance of siloed BEAN:ETH from uniswap pool
+    await unwrapUniswapLPs(balances, lpPositions, block);
+
+    // add balances of all siloed curve pools
+    // this is the block when SiloV2Facet with getTotalDeposited() was introduced
+    if (block >= 14218934) {
+        await Promise.all(BEAN_CRV_POOLS.map(async (pool) => {
+            const lpBalance = (await sdk.api.abi.call({
+                abi: bean_abi["getTotalDeposited"],
```
Comment (g1nt0ki) [inline projects/bean/index.js:51]: can you also replace the calls here with erc20.balanceOf calls.

hmmm, I see that the curve pools have bean tokens in them, let me check with others if these too must go under pool2

**Review by g1nt0ki — CHANGES_REQUESTED** [review]
The code is good, just a minor change on how you fetch token balances.

Also, please take out pool2 and staking part as their own exports.

**Comment by llamatester** [comment]
The adapter at projects/bean/index.js exports TVL: 
        
 
 
    ethereum                  0
    ethereum-staking          30.69 M
    ethereum-pool2            138.82 M
    staking                   30.69 M
    pool2                     138.82 M
    
    total                    0

**Pull request merged**